File: Algorithms/powx_n.cpp
#include "leetcode.h"
class Solution {
public:
    double myPow(double x, int n) {
        double result = 1.0;
        if (n == 0)
            return 1.0;
        bool positive = true;
        if (n < 0) {
            positive = false;
            n = -n;
        }
        while (n != 0) {
            if (n%2 == 1) {
                result *= x;
            }
            x *= x;
            n /= 2;
        }
        if (positive) {
            return result;
        }
        else {
            return 1/result;
        }
    }
};

File: Algorithms/powx_n.py
class Solution:
    # @param {float} x
    # @param {integer} n
    # @return {float}
    def myPow(self, x, n):
        if n == 0:
            return 1.0
        positive = True
        if n < 0:
            positive = False
            n = -n
        result = 1.0
        while n:
            if n % 2: result *= x
            n /= 2
            x *= x
        return result if positive else 1/result
